nodejs/lancedb/embedding/embedding_function.ts

```diff
@@ -1,3 +1,7 @@
+import { DataType, Field, FixedSizeList, Float, Float32 } from "apache-arrow";
+import "reflect-metadata";
```

Comment:
What is this?

Reply:
It's a pretty commonly used library when working with metadata and decorators. 

It allows us to set some metadata on the class prototype. It's used to add the function name to the embedding function if you register it using the decorator. 

such as 

```ts
@register("my-function")
class MyEmbeddingFunction extends EmbeddingFunction<string> {
// ...
}
```
then the `register` function injects some metadata into the class prototype
https://github.com/universalmind303/lancedb/blob/90afcc051d35d4d7edc7cdaa832ae353670afe14/nodejs/lancedb/embedding/registry.ts#L51

which we can later pull back off the class when we convert the embedding functions to metadata 
https://github.com/universalmind303/lancedb/blob/90afcc051d35d4d7edc7cdaa832ae353670afe14/nodejs/lancedb/embedding/registry.ts#L115